### PR TITLE
Loaders

### PR DESCRIPTION
**Pull request by SBub**
Addresses: https://www.notion.so/jolocom/Create-a-loader-module-80d646f66a4b4632a43231e85ebcf17e

Loader module:
* holds the state of loader: type and msg
* useLoaderScreenVisibility hook listens for updates in the loader module and shows/hides Loader screen accordingly

### CODE REVIEW

File: src/types/loader.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export enum LoaderMsgs {
+  creating = 'Creating your personal secret number',
+  matching = 'Matching two instances',
+  success = 'Success!',
+  failed = 'Failed',
+  empty = '',
+}
```
Comment (clauxx) [inline src/types/loader.ts:7]: I think this shouldn't really be an `enum`. They can go to the `~/translations/strings.ts` file after the Error Boundary PR is merged.

File: src/modules/loader/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export const getLoaderType = createSelector(
+  getLoaderState,
+  (loader) => loader.type,
+);
+
+export const getLoaderMsg = createSelector(
+  getLoaderState,
+  (loader) => loader.msg,
+);
```
Comment (clauxx) [inline src/modules/loader/selectors.ts:17]: any reason to use reselect's `createSelector()`? The selectors are not doing any transformation on the state so far, so I see no reason to memoize them, but I may be missing something.

Comment (SBub) [reply src/modules/loader/selectors.ts:17]: That's actually true

File: src/RootNavigation.tsx [hunk 3]
```diff
@@ -10,11 +11,53 @@ import Interactions from '~/screens/Modals/Interactions';
...
+  [ScreenNames.LoggedOut]: undefined;
+  [ScreenNames.LoggedIn]: undefined;
+  [ScreenNames.Loader]: undefined;
+  [ScreenNames.Interactions]: undefined;
+};
+
+const RootStack = createStackNavigator<RootStackParamList>();
+
+// a listener for loader module state
+const useLoaderScreenVisibility = () => {
```
[the remark this answers is not in the document]
Comment (SBub) [reply src/RootNavigation.tsx:26]: it isn't reused, it is just specific to this file

File: src/modules/loader/reducers.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import {LoaderStateI, LoaderMsgs} from '~/types/loader';
+
+import {LoaderActions, LoaderTypes} from '~/types/loader';
```
[the remark this answers is not in the document]
Comment (SBub) [reply src/modules/loader/reducers.ts:3]: The reason I moved it out of modules is that it was used outside of the modules folder. i.e. `LoaderTypes` is used in `Claims` screen.

[the remark this answers is not in the document]
Comment (SBub) [reply src/modules/loader/reducers.ts:3]: Definitely joining

File: src/types/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import {LoaderStateI} from '~/types/loader';
+
+export interface RootReducerI {
+  loader: LoaderStateI;
+}
```
[the remark this answers is not in the document]
Comment (SBub) [reply src/types/reducer.ts:5]: Yes this one is not reused, can sit in the same file

File: src/translations/strings.ts [hunk 6]
```diff
@@ -5,3 +5,11 @@ export const strings = {
...
   CLOSE: 'Close',
 };
+
+export const LoaderMsgs = {
+  CREATING: 'Creating your personal secret number',
+  MATCHING: 'Matching two instances',
+  SUCCESS: 'Success!',
+  FAILED: 'Failed',
+  EMPTY: '',
+};
```
Comment (clauxx) [inline src/translations/strings.ts:15]: We try to put all the text we use within the app into the strings object to make it easier to export it for translations. Check out the old repo for reference. `LoaderMsgs` can still be a separate object, but we should spread it into `strings` and only export `strings`. We won't use translations for now, but when we will, it'll be useful.

**Pull request merged**